network-area-diagram/src/main/java/com/powsybl/nad/NadParameters.java

```diff
@@ -0,0 +1,74 @@
+/**
+ * Copyright (c) 2023, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.powsybl.nad;
+
+import com.powsybl.iidm.network.Network;
+import com.powsybl.nad.layout.LayoutFactory;
+import com.powsybl.nad.layout.LayoutParameters;
+import com.powsybl.nad.svg.LabelProvider;
+import com.powsybl.nad.svg.SvgParameters;
+import com.powsybl.nad.svg.iidm.IdProviderFactory;
+import com.powsybl.nad.svg.iidm.LabelProviderFactory;
+import com.powsybl.nad.svg.iidm.StyleProviderFactory;
+
+import java.util.Objects;
+
+/**
+ *
+ * @author Sophie Frasnedo <sophie.frasnedo at rte-france.com>
+ */
+public class NadParameters {
+
+    private final SvgParameters svgParameters;
+    private final LayoutParameters layoutParameters;
+    private final StyleProviderFactory styleProviderFactory;
+    private final LabelProviderFactory labelProviderFactory;
+    private final LayoutFactory layoutFactory;
+    private final IdProviderFactory idProviderFactory;
+
+    public NadParameters(SvgParameters svgParameters, LayoutParameters layoutParameters, StyleProviderFactory styleProviderFactory, LabelProviderFactory labelProviderFactory, LayoutFactory layoutFactory, IdProviderFactory idProviderFactory) {
+        this.svgParameters = Objects.requireNonNull(svgParameters);
+        this.layoutParameters = Objects.requireNonNull(layoutParameters);
+        this.styleProviderFactory = Objects.requireNonNull(styleProviderFactory);
+        this.labelProviderFactory = Objects.requireNonNull(labelProviderFactory);
+        this.layoutFactory = Objects.requireNonNull(layoutFactory);
+        this.idProviderFactory = Objects.requireNonNull(idProviderFactory);
+    }
+
+    public SvgParameters getSvgParameters() {
+        return svgParameters;
+    }
+
+    public LayoutParameters getLayoutParameters() {
+        return layoutParameters;
+    }
+
+    public StyleProviderFactory getStyleProviderFactory() {
+        return styleProviderFactory;
+    }
+
+    public LabelProvider createLabelProvider(Network network) {
+        return labelProviderFactory.create(network, svgParameters);
```

Comment:
Should should avoid this kind of method that does nothing really helpful
```java
parameters.getLabelProvider().create()
```
is enough.

Reply:
This method was created so that the svgParameters call is masked and dealt with inside the NadParameters class. There is not any possibility for a user to feed the create function with a SvgParameters that is not the 
It "replaces" the getter. There is no getter for a LabelProviderFactory or a LabelProvider in the NadParameters class (it may be wrong :)). 
What do you think?